lib/IRGen/IRGenDebugInfo.cpp

```diff
@@ -2675,7 +2682,15 @@ void IRGenDebugInfoImpl::addFailureMessageToCurrentLoc(IRBuilder &Builder,
   assert(parentScopesAreSane(TrapSc) && "parent scope sanity check failed");
 
   // Wrap the existing TrapLoc into the failure function.
-  auto DL = llvm::DILocation::get(IGM.getLLVMContext(), 0, 0, TrapSP, TrapLoc);
+  // Line 0 is invalid in CodeView, so use the line and column from the original
+  // trap location.
+  auto DL = llvm::DILocation::get(
+      IGM.getLLVMContext(),
+      Opts.DebugInfoFormat == IRGenDebugInfoFormat::CodeView ? TrapLoc.getLine()
+                                                             : 0,
+      Opts.DebugInfoFormat == IRGenDebugInfoFormat::CodeView ? TrapLoc.getCol()
+                                                             : 0,
+      TrapSP, TrapLoc);
```

Comment:
This will likely result in larger debug info, since the abstract origin of the inlined function can no longer be uniqued but that's the kind of tradeoff you'll need to make.

Reply:
Yep, I agree. I think the win in crash reporting investigation is worth it here.